lib/flutter/flutter_tools.dart

```diff
@@ -78,19 +83,35 @@ class FlutterToolsManager implements Disposable {
       return;
     }
 
-    DartProject project = projectManager.getProjectFor(editor?.getPath());
+    DartProject project = projectManager.getProjectFor(editor.getPath());
     if (project == null) {
       atom.notifications.addWarning('The current project is not a Dart project.');
       return;
     }
 
     FlutterTool flutter = _flutterSdk.sdk.flutterTool;
-    flutter.runInJob(
-      ['upgrade'],
+    flutter.runInJob(['upgrade'],
       title: 'Running Flutter upgrade…',
       cwd: project.directory.path
     );
   }
 
+  void _doctor(AtomEvent _) {
+    if (!_flutterSdk.hasSdk) {
+      _flutterSdk.showInstallationInfo();
+      return;
+    }
+
+    FlutterTool flutter = _flutterSdk.sdk.flutterTool;
+
+    TextEditor editor = atom.workspace.getActiveTextEditor();
+    DartProject project = projectManager.getProjectFor(editor?.getPath());
+
+    flutter.runInJob(['doctor'],
+      title: 'Running Flutter doctor…',
+      cwd: project?.directory?.path
```

Comment:
OK to run this outside a flutter project? Good by me if that makes sense. Otherwise might want to not run flutter doctor, disable action, show notification, or something like that.